src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,43 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  active: boolean;
+  selectedUserId: number | null;
+  onClose: () => void;
+  selectedTodo: Todo | null;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  active,
+  selectedUserId,
+  onClose,
+  selectedTodo,
+}) => {
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (selectedUserId !== null) {
+      setLoading(true);
+      getUser(selectedUserId)
+        .then(data => setUser(data))
+        .finally(() => setLoading(false));
+    }
+  }, [selectedUserId]);
+
+  if (!active) {
+    return null;
+  }
 
-export const TodoModal: React.FC = () => {
   return (
-    <div className="modal is-active" data-cy="modal">
+    <div className={active ? 'modal is-active' : 'modal'} data-cy="modal">
```

Comment:
Consider using the `classnames` library for managing conditional class names here. This will align with the task requirements and improve code readability.